src/scripts.c

```diff
@@ -122,7 +117,17 @@ void run_preupdate_scripts(struct manifest *manifest)
 	char *script;
 	__attribute__((unused)) int ret = 0;
 
-	string_or_die(&script, "/usr/bin/clr_pre_update.sh");
+	if (strlen(PRE_UPDATE) == 0) {
+		return;
+	}
+
+	if (strcmp("/", path_prefix) == 0) {
```

Comment:
This took me a minute.

So if there is no `path_prefix` specified (other than root) the pre-update script is expected to run against the current root, yet if there is a `path_prefix` specified the script is expected to live under the `path_prefix` and then also needs that passed as an argument?

Can this be cleaned up so that it either always takes the argument or never does (that might be harder).

If not, the requirements for these should be well-documented somewhere.

Reply:
Well it would be hard for the script to know to run against the prefix if the prefix wasn't passed to it. We could have a heuristic that the script could do to check if it is in /usr/bin to decide if it is prefix mode or not but that seems like a bad idea.

Again the script could just decide to not handle this case at all or do something different and the OSV would need to have a good idea of how they want them to work. For now I'm fine with documenting the options required wherever you think is best. Really though these are all better off being inside of swupd proper as plugins or something.

Reply:
Being inside of swupd is the right solution, you are right.